cmd/dockerd/daemon.go

```diff
@@ -277,6 +286,7 @@ func (cli *DaemonCli) start(opts daemonOptions) (err error) {
 	errAPI := <-serveAPIWait
 	c.Cleanup()
 	shutdownDaemon(d)
+	close(watchStream)
```

Comment:
There shouldn't be a need to close this. I think there is a possibility that cluster can still attempt to write to this chan after `Cleanup` has been called(although very unlikely as `shutdownDaemon` is not fast).